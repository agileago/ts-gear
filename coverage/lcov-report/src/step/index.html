
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/step</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> src/step</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.24% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>195/290</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">49.17% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>59/120</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">62.07% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>36/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">67.02% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>191/285</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="assembleSchemaToGlobal.ts"><a href="assembleSchemaToGlobal.ts.html">assembleSchemaToGlobal.ts</a></td>
	<td data-value="96.55" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 96%"></div><div class="cover-empty" style="width: 4%"></div></div>
	</td>
	<td data-value="96.55" class="pct high">96.55%</td>
	<td data-value="29" class="abs high">28/29</td>
	<td data-value="91.67" class="pct high">91.67%</td>
	<td data-value="12" class="abs high">11/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="96.43" class="pct high">96.43%</td>
	<td data-value="28" class="abs high">27/28</td>
	</tr>

<tr>
	<td class="file medium" data-value="cleanRefAndDefinitionName.ts"><a href="cleanRefAndDefinitionName.ts.html">cleanRefAndDefinitionName.ts</a></td>
	<td data-value="78.57" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 78%"></div><div class="cover-empty" style="width: 22%"></div></div>
	</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="6" class="abs high">5/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="14" class="abs medium">11/14</td>
	</tr>

<tr>
	<td class="file high" data-value="colletRefsInRequestAndPatchDefinition.ts"><a href="colletRefsInRequestAndPatchDefinition.ts.html">colletRefsInRequestAndPatchDefinition.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="6" class="abs high">6/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="23" class="abs high">23/23</td>
	</tr>

<tr>
	<td class="file high" data-value="fetchSwagger.ts"><a href="fetchSwagger.ts.html">fetchSwagger.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	</tr>

<tr>
	<td class="file low" data-value="generateDefinitionContent.ts"><a href="generateDefinitionContent.ts.html">generateDefinitionContent.ts</a></td>
	<td data-value="19.35" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 19%"></div><div class="cover-empty" style="width: 81%"></div></div>
	</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="19.35" class="pct low">19.35%</td>
	<td data-value="31" class="abs low">6/31</td>
	</tr>

<tr>
	<td class="file low" data-value="importAllDefinition.ts"><a href="importAllDefinition.ts.html">importAllDefinition.ts</a></td>
	<td data-value="22.22" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 22%"></div><div class="cover-empty" style="width: 78%"></div></div>
	</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	</tr>

<tr>
	<td class="file high" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="11" class="abs high">11/11</td>
	</tr>

<tr>
	<td class="file high" data-value="parseGenericType.ts"><a href="parseGenericType.ts.html">parseGenericType.ts</a></td>
	<td data-value="98.63" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 98%"></div><div class="cover-empty" style="width: 2%"></div></div>
	</td>
	<td data-value="98.63" class="pct high">98.63%</td>
	<td data-value="73" class="abs high">72/73</td>
	<td data-value="86.84" class="pct high">86.84%</td>
	<td data-value="38" class="abs high">33/38</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="19" class="abs high">19/19</td>
	<td data-value="98.57" class="pct high">98.57%</td>
	<td data-value="70" class="abs high">69/70</td>
	</tr>

<tr>
	<td class="file medium" data-value="prepareProjectDirectory.ts"><a href="prepareProjectDirectory.ts.html">prepareProjectDirectory.ts</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

<tr>
	<td class="file low" data-value="translateSchema.ts"><a href="translateSchema.ts.html">translateSchema.ts</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="45" class="abs low">9/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="24" class="abs low">0/24</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="20.45" class="pct low">20.45%</td>
	<td data-value="44" class="abs low">9/44</td>
	</tr>

<tr>
	<td class="file low" data-value="writeProject.ts"><a href="writeProject.ts.html">writeProject.ts</a></td>
	<td data-value="35.71" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 35%"></div><div class="cover-empty" style="width: 65%"></div></div>
	</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="28" class="abs low">10/28</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Apr 27 2020 18:43:36 GMT+0800 (GMT+08:00)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    