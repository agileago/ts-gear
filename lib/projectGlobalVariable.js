"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const clearObject_1 = require("./tool/clearObject");
const projectGlobal = {};
exports.getGlobal = (project) => {
    if (!projectGlobal[project.name]) {
        projectGlobal[project.name] = {
            definitionMap: {},
            requestMap: {},
            requestRefSet: new Set(),
        };
    }
    return projectGlobal[project.name];
};
exports.restore = (project) => {
    const g = projectGlobal[project.name];
    clearObject_1.clearObject(g.definitionMap);
    clearObject_1.clearObject(g.requestMap);
    g.requestRefSet.clear();
};
exports.httpMethods = ['get', 'put', 'post', 'delete', 'options', 'head', 'patch'];
