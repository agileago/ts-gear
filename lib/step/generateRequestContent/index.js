"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateRequestContent = void 0;
const os_1 = require("os");
const source_1 = require("../../source");
const transformSwaggerPathToRouterPath_1 = require("../../tool/transformSwaggerPathToRouterPath");
const projectGlobalVariable_1 = require("../../projectGlobalVariable");
const assembleDoc_1 = require("../../tool/assembleDoc");
const constant_1 = require("../../constant");
const generateMockData_1 = require("./generateMockData");
const generateResponseType_1 = require("./generateResponseType");
const generateParameterType_1 = require("./generateParameterType");
const join = require("url-join");
/** from swagger spec paths assemble request functions */
exports.generateRequestContent = (spec, project) => {
    const { pathMatcher, withBasePath, withHost } = project;
    const { requestMap, definitionMap, enumMap } = projectGlobalVariable_1.getGlobal(project);
    const shouldMockResponseStatement = project.shouldMockResponseStatement || constant_1.defaultShouldMockResponseStatement;
    const resultContent = [];
    Object.getOwnPropertyNames(requestMap).forEach((requestFunctionName) => {
        const requestTypeScriptContent = [];
        const request = requestMap[requestFunctionName];
        const { httpMethod } = request;
        if (pathMatcher) {
            if (typeof pathMatcher === 'function') {
                if (!pathMatcher(request.pathName, httpMethod)) {
                    return;
                }
            }
            else if (!pathMatcher.test(request.pathName)) {
                return;
            }
        }
        let parameterTypeName = '';
        let parameterRequired = false;
        if (request.parameters && request.parameters.length > 0) {
            const parameterType = generateParameterType_1.generateParameterType(requestFunctionName, request.parameters);
            parameterTypeName = parameterType.parameterTypeName;
            parameterRequired = parameterType.parameterRequired;
            requestTypeScriptContent.push(parameterType.parameterTypeContent);
        }
        const responseType = generateResponseType_1.generateResponseType(requestFunctionName, request.responses);
        requestTypeScriptContent.push(responseType.responseTypeContent);
        requestTypeScriptContent.push(responseType.successTypeContent);
        const urlPath = join(spec.basePath || '', transformSwaggerPathToRouterPath_1.transformSwaggerPathToRouterPath(String(request.pathName)));
        const source = source_1.sow();
        const requesterStatment = `return requester('${urlPath}', {${withHost ? `, host: '${spec.host}'` : ''}${withBasePath ? `, basePath: '${spec.basePath}'` : ''}method: '${httpMethod}'${parameterTypeName ? ', ...option' : ''}}) as Promise<any>`;
        const functionStatment = `if (${shouldMockResponseStatement}) {
      return Promise.resolve(${requestFunctionName}.mockData as any)
    }
    ${requesterStatment}`;
        const functionData = {
            name: requestFunctionName,
            isExported: true,
            returnType: `Promise<${responseType.successTypeName}>`,
            statements: functionStatment,
            docs: assembleDoc_1.assembleDoc(request.schema),
        };
        functionData.parameters = [];
        if (parameterTypeName) {
            functionData.parameters.push({
                hasQuestionToken: !parameterRequired,
                name: 'option',
                type: parameterTypeName,
            });
        }
        source.addFunction(functionData);
        const mockStatment = `${requestFunctionName}.mockData = (${JSON.stringify(generateMockData_1.generateMockData(request, definitionMap, enumMap))} as any)`;
        source.addStatements(`
if (${shouldMockResponseStatement}) {
  ${mockStatment}
}
${requestFunctionName}.method = '${httpMethod}'
`);
        requestTypeScriptContent.push(source_1.harvest(source));
        /** store typescript content to requestMap */
        request.typescriptContent = requestTypeScriptContent.join(os_1.EOL);
        resultContent.push(request.typescriptContent);
    });
    /** return value only for test and debug */
    return resultContent.join(os_1.EOL);
};
