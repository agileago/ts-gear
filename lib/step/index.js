"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./getUserConfig"));
__export(require("./fetchSwagger"));
__export(require("./prepareProjectDirectory"));
__export(require("./translateSchema"));
__export(require("./assembleSchemaToGlobal"));
__export(require("./cleanRefAndDefinitionName"));
__export(require("./parseGenericType"));
__export(require("./colletRefsInRequestAndPatchDefinition"));
__export(require("./generateDefinitionContent"));
__export(require("./generateRequestContent"));
__export(require("./writeProject"));
