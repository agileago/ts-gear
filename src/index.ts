export { requester as fetchRequester } from './requester/fetch'
export { requester as axiosRequester } from './requester/axios'
export { IProject, PropertyType } from './interface'
