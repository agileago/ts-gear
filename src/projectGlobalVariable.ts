import type { ProjectGlobalMap, Project } from './type'
import { clearObject } from './tool/clearObject'

const projectGlobal: ProjectGlobalMap = {}

// let currentProject: Project | undefined

export const getGlobal = (project: Project) => {
  if (!projectGlobal[project.name]) {
    projectGlobal[project.name] = {
      definitionMap: {},
      requestMap: {},
      requestRefSet: new Set(),
      requestEnumSet: new Set(),
      enumMap: {},
    }
  }
  return projectGlobal[project.name]
}

export const restore = (project: Project) => {
  const g = projectGlobal[project.name]
  clearObject(g.definitionMap)
  clearObject(g.requestMap)
  clearObject(g.enumMap)
  g.requestRefSet.clear()
  g.requestEnumSet.clear()
}
