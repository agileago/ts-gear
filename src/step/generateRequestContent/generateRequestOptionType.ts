import type { Parameter, Reference } from 'swagger-schema-official'
import { isEmpty, upperFirst } from 'lodash'
import type { RequestParameterPosition, Project } from '../../type'
import { schemaToTypescript } from '../../tool/schemaToTypescript'
import { sow, harvest } from '../../source'
import { assembleRequestParam } from './assembleRequestParam'

/**
 * @param name request function parameter interface name
 * @param parameters swagger request parameters
 * */
export const generateRequestOptionType = (
  functionName: string,
  parameters: Array<Parameter | Reference>,
  project: Project,
) => {
  const source = sow()
  const parameterTypeName = `${upperFirst(functionName)}Option`
  const inter = source.addInterface({
    isExported: project.shouldExportRequestOptionType === undefined || !!project.shouldExportRequestOptionType,
    name: parameterTypeName,
    docs: [`@description request parameter type for ${functionName}`],
  })
  const assembledParameters = assembleRequestParam(parameters)
  let parameterRequired = false
  ;(Object.getOwnPropertyNames(assembledParameters) as RequestParameterPosition[]).forEach(position => {
    const param = assembledParameters[position]!
    if (!parameterRequired) {
      parameterRequired = !isEmpty(param.required)
    }
    inter.addProperty({
      name: position,
      type: schemaToTypescript(param),
      hasQuestionToken: isEmpty(param.required),
      docs: param.docs,
    })
  })
  return {
    parameterTypeName,
    parameterTypeContent: harvest(source),
    parameterRequired,
  }
}
