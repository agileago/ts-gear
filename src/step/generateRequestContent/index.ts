import { EOL } from 'os'

import { FunctionDeclarationStructure, OptionalKind } from 'ts-morph'
import { Spec } from 'swagger-schema-official'

import { IProject } from '../../interface'
import { sow, harvest } from '../../source'
import { transformSwaggerPathToRouterPath } from '../../tool/transformSwaggerPathToRouterPath'
import { getGlobal } from '../../projectGlobalVariable'
import { assembleDoc } from '../../tool/assembleDoc'
import { defaultShouldMockResponseStatement } from '../../constant'

import { generateMockData } from './generateMockData'
import { generateResponseType } from './generateResponseType'
import { generateParameterType } from './generateParameterType'

import join = require('url-join')

/** from swagger spec paths assemble request functions */
export const generateRequestContent = (spec: Spec, project: IProject) => {
  const { pathMatcher, withBasePath, withHost } = project
  const { requestMap, definitionMap, enumMap } = getGlobal(project)

  const shouldMockResponseStatement = project.shouldMockResponseStatement || defaultShouldMockResponseStatement

  const resultContent: string[] = []
  Object.getOwnPropertyNames(requestMap).forEach((requestFunctionName) => {
    const requestTypeScriptContent: string[] = []
    const request = requestMap[requestFunctionName]
    const { httpMethod } = request
    if (pathMatcher) {
      if (typeof pathMatcher === 'function') {
        if (!pathMatcher(request.pathName, httpMethod)) {
          return
        }
      } else if (!pathMatcher.test(request.pathName)) {
        return
      }
    }

    let parameterTypeName = ''
    let parameterRequired = false
    if (request.parameters && request.parameters.length > 0) {
      const parameterType = generateParameterType(requestFunctionName, request.parameters)
      parameterTypeName = parameterType.parameterTypeName
      parameterRequired = parameterType.parameterRequired
      requestTypeScriptContent.push(parameterType.parameterTypeContent)
    }
    const responseType = generateResponseType(requestFunctionName, request.responses)
    requestTypeScriptContent.push(responseType.responseTypeContent)
    requestTypeScriptContent.push(responseType.successTypeContent)
    const urlPath = join(spec.basePath || '', transformSwaggerPathToRouterPath(String(request.pathName)))
    const source = sow()
    const requesterStatment = `return requester('${urlPath}', {${withHost ? `, host: '${spec.host}'` : ''}${
      withBasePath ? `, basePath: '${spec.basePath}'` : ''
    }method: '${httpMethod}'${parameterTypeName ? ', ...option' : ''}}) as Promise<any>`
    const functionStatment = `if (${shouldMockResponseStatement}) {
      return Promise.resolve(${requestFunctionName}.mockData as any)
    }
    ${requesterStatment}`
    const functionData: OptionalKind<FunctionDeclarationStructure> = {
      name: requestFunctionName,
      isExported: true,
      returnType: `Promise<${responseType.successTypeName}>`,
      statements: functionStatment,
      docs: assembleDoc(request.schema),
    }
    functionData.parameters = []
    if (parameterTypeName) {
      functionData.parameters.push({
        hasQuestionToken: !parameterRequired,
        name: 'option',
        type: parameterTypeName,
      })
    }
    source.addFunction(functionData)
    const mockStatment = `${requestFunctionName}.mockData = (${JSON.stringify(
      generateMockData(request, definitionMap, enumMap),
    )} as any)`
    source.addStatements(`
if (${shouldMockResponseStatement}) {
  ${mockStatment}
}
${requestFunctionName}.method = '${httpMethod}'
`)
    requestTypeScriptContent.push(harvest(source))
    /** store typescript content to requestMap */
    request.typescriptContent = requestTypeScriptContent.join(EOL)
    resultContent.push(request.typescriptContent)
  })

  /** return value only for test and debug */
  return resultContent.join(EOL)
}
