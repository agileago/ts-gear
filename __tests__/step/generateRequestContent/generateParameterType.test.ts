import type { Parameter, Spec } from 'swagger-schema-official'
import { cloneDeep } from 'lodash'
import { generateRequestOptionType } from 'src/step/generateRequestContent/generateRequestOptionType'
import * as petSpec from 'example/fixture/pet.json'
import * as step from 'src/step'
import { restore } from 'src/projectGlobalVariable'
import type { Project } from 'src/type'

describe('src/step/generateRequestContent/generateParameterType', () => {
  const project: Project = {
    name: 'pet',
    dest: './service',
    source: 'fixture/pet.json',
    importRequesterStatement: 'import { requester } from "ts-gear/requester/fetch"',
  }
  it('generateParameterType case 1', () => {
    const spec = cloneDeep(petSpec) as Spec
    step.cleanRefAndDefinitionName(spec, true)
    step.assembleSchemaToGlobal(spec, project)
    const content = generateRequestOptionType(
      'postUserCreateWithList',
      spec.paths['/user/createWithList'].post!.parameters as Parameter[],
      project,
    )
    expect(content).toMatchSnapshot()
    restore(project)
  })

  it('generateParameterType case 2', () => {
    const spec = cloneDeep(petSpec) as Spec
    step.cleanRefAndDefinitionName(spec, true)
    step.assembleSchemaToGlobal(spec, project)
    const content = generateRequestOptionType(
      'postUserCreateWithList',
      spec.paths['/user/createWithList'].post!.parameters as Parameter[],
      project,
    )
    expect(content).toMatchSnapshot()
    restore(project)
  })
})
