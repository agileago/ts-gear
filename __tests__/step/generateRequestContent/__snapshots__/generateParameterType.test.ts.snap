// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/step/generateRequestContent/generateParameterType generateParameterType case 1 1`] = `
Object {
  "parameterRequired": true,
  "parameterTypeContent": "/** @description request parameter type for postUserCreateWithList */
export interface PostUserCreateWithListOption {
    /**
     * @description
     *   List of user object
     */
    body: Array<User>;
}
",
  "parameterTypeName": "PostUserCreateWithListOption",
}
`;

exports[`src/step/generateRequestContent/generateParameterType generateParameterType case 2 1`] = `
Object {
  "parameterRequired": true,
  "parameterTypeContent": "/** @description request parameter type for postUserCreateWithList */
export interface PostUserCreateWithListOption {
    /**
     * @description
     *   List of user object
     */
    body: Array<User>;
}
",
  "parameterTypeName": "PostUserCreateWithListOption",
}
`;
